### PR TITLE
Dev/string fix

### PR DESCRIPTION
**Pull request by Nordgaren**
Fixes strings being added multiple times (including null string)

### CODE REVIEW

File: FSParam/Param.cs [hunk 1]
```diff
@@ -897,26 +897,37 @@ protected override void Write(BinaryWriterEx bw)
             }
             
             // Write row names
-            for (int i = 0; i < Rows.Count; i++)
+            Dictionary<string, long> stringOffsetDictionary = new Dictionary<string, long> {
```
Comment (katalash) [inline FSParam/Param.cs:900]: I'd rather you use DSMS style conventions ( "{" always on a new line)

File: SoulsFormats/SoulsFormats/Formats/PARAM/PARAM/PARAM.cs [hunk 2]
```diff
@@ -202,11 +202,19 @@ protected override void Write(BinaryWriterEx bw)
                 bw.FillInt64("ParamTypeOffset", bw.Position);
                 bw.WriteASCII(ParamType, true);
             }
-
+            
+            StringOffsetDictionary = new Dictionary<string, long>() {
```
Comment (katalash) [inline SoulsFormats/SoulsFormats/Formats/PARAM/PARAM/PARAM.cs:206]: same

File: SoulsFormats/SoulsFormats/Formats/PARAM/PARAM/Row.cs [hunk 3]
```diff
@@ -326,20 +326,24 @@ internal void WriteCells(BinaryWriterEx bw, PARAM parent, int index)
 
             internal void WriteName(BinaryWriterEx bw, PARAM parent, int i)
             {
-                long nameOffset = 0;
-                if (Name != null)
+                if (Name == null) Name = string.Empty;
```
Comment (katalash) [inline SoulsFormats/SoulsFormats/Formats/PARAM/PARAM/Row.cs:329]: I also tend to prefer conditionals having their code on a new line as well

**Comment by Nordgaren** [comment]
Fixed all of these issues. LMK if there's any more changes you'd like me to make. :)

**Pull request merged**